pkg/workloadapi/workload.proto

```diff
@@ -215,6 +216,33 @@ enum TunnelProtocol {
   // Future options may include things like QUIC/HTTP3, etc.
 }
 
+// NativeTunnel describes how an application or gateway
+// can consume tunnel information directly.
+message NativeTunnel {
+  // A target natively handles this type of traffic.  
+  GatewayProtocol protocol;
+  // optional: if set, traffic should be sent to this port after the last zTunnel hop
+  uint32 port;
+}
+
+enum GatewayProtocol {
+  // Bytes are just copied directly.
+  NONE = 1;
+
+  // The workload nativley supports HBONE
+  // For the zTunnel, this means:
+  // * Requests *from* this workload do not need to be tunneled if they already are tunneled by the tunnel_protocol.
+  // * Requests *to* this workload, via the tunnel_protocol, do not need to be de-tunneled.
+  HBONE = 2;
+
+  // Prepend PROXY protocol headers before copying bytes
+  // Standard PROXY source and destination information
+  // is included, along with potential extra TLV headers:
+  // 0xD0 - The SPIFFE identity of the source workload
+  // 0xD1 - The FQDN or Hostname of the targeted Service
+  PROXY = 3;
```

Comment:
HBONE implies something about transport security.

PROXY does not.

Do we want to make this more explicit by doing something like

PROXY_HBONE?

Reply:
proxy can be over tls

Reply:
Right - HBONE implies mTLS. Proxy implies...TLS, mTLS, or really anything. That's the bit that's a bit confusing here.

Reply:
So I added HBONE in here in an attempt to consolidate the "native tunneled-ness". In hindsight I think this field should purely talk about what is on the _inbound_  stream. 

TunnelProtocol will talk about outbound.